raphtory/src/io/parquet_loaders.rs

```diff
@@ -0,0 +1,315 @@
+        if let Some(ref layer) = layer {
+            cols_to_check.push(layer.as_ref());
+        }
+    }
+
+    let df = process_parquet_file_to_df(parquet_file_path, cols_to_check.clone())?;
+    df.check_cols_exist(&cols_to_check)?;
+    let size = cols_to_check.len();
+
+    load_edges_deletions_from_df(
+        &df,
+        size,
+        src,
+        dst,
+        time,
+        layer,
+        layer_in_df.unwrap_or(true),
+        graph,
+    )
+    .map_err(|e| GraphError::LoadFailure(format!("Failed to load graph {e:?}")))?;
+
+    Ok(())
+}
+
+pub(crate) fn process_parquet_file_to_df(
+    parquet_file_path: &Path,
+    col_names: Vec<&str>,
+) -> Result<PretendDF, GraphError> {
+    let (names, arrays) = read_parquet_file(parquet_file_path, &col_names)?;
+
+    let names = names
+        .into_iter()
+        .filter(|x| col_names.contains(&x.as_str()))
+        .collect();
+    let arrays = arrays
+        .map_ok(|r| r.into_iter().map(|boxed| boxed.clone()).collect_vec())
+        .collect::<Result<Vec<_>, _>>()?;
+
+    Ok(PretendDF { names, arrays })
+}
+
+fn read_parquet_file(
+    path: impl AsRef<Path>,
+    col_names: &Vec<&str>,
+) -> Result<
+    (
+        Vec<String>,
+        impl Iterator<Item = Result<Chunk<Box<dyn Array>>, error::PolarsError>>,
+    ),
+    GraphError,
+> {
+    let read_schema = |metadata: &FileMetaData| -> Result<ArrowSchema, GraphError> {
+        let schema = read::infer_schema(metadata)?;
+        let fields = schema
+            .fields
+            .iter()
+            .map(|f| {
+                if f.data_type == DataType::Utf8View {
+                    Field::new(f.name.clone(), DataType::LargeUtf8, f.is_nullable)
+                } else {
+                    f.clone()
+                }
+            })
+            .filter(|f| {
+                // Filtered fields to avoid loading data that is not needed
+                col_names.contains(&f.name.as_str())
+            })
+            .collect::<Vec<_>>();
+
+        Ok(ArrowSchema::from(fields).with_metadata(schema.metadata))
+    };
+
+    let mut file = std::fs::File::open(&path)?;
+    let metadata = read_metadata(&mut file)?;
+    let row_groups = metadata.clone().row_groups;
+    let schema = read_schema(&metadata)?;
+
+    // Although fields are already filtered by col_names, we need names in the order as it appears
+    // in the schema to create PretendDF
+    let names = schema.fields.iter().map(|f| f.name.clone()).collect_vec();
+
+    let reader = FileReader::new(file, row_groups, schema, None, None, None);
+    Ok((names, reader))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use polars_arrow::array::{PrimitiveArray, Utf8Array};
+    use std::path::PathBuf;
+
+    #[test]
+    fn test_process_parquet_file_to_df() {
+        let parquet_file_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .parent()
+            .map(|p| p.join("raphtory/resources/test/test_data.parquet"))
```

Comment:
What is up with this file path? It looks like the call to `.parent` should not be necessary?